service/history/api/multioperation/api.go

```diff
@@ -192,7 +192,7 @@ func Invoke(
 		case enumspb.WORKFLOW_ID_CONFLICT_POLICY_UNSPECIFIED:
 			// ... fail since this policy is invalid
 			currentWorkflowLease.GetReleaseFn()(nil) // nil since nothing was modified
-			return nil, serviceerror.NewInternal("unhandled workflow id conflict policy: unspecified")
+			return nil, serviceerror.NewUnimplemented("unhandled workflow id conflict policy: unspecified")
 		}
 	}
```

Comment:
I didn't understand the meaning of `Internal` when I wrote this initially; these should actually *not* be retried.

Reply:
We don't retry internal by default (in most/all places?) I know, it confuses many devs.

Isn't this an invalid argument though?

Reply:
But we do retry Internal for IsServiceHandlerRetryableError and IsServiceClientTransientError... I think agree we probably shouldn't but in in the conversation about this a few weeks ago, I think the conclusion was to not change anything for now

Reply:
👍 Right, we do retry it.

<img width="599" alt="Screenshot 2024-12-03 at 5 08 57 PM" src="https://github.com/user-attachments/assets/a4933ed5-751a-4e13-8a34-f1e599eb4e6a">

Reply:
> Isn't this an invalid argument though?

I can make it invalid argument.

Reply:
> we do retry Internal for IsServiceHandlerRetryableError and IsServiceClientTransientError

I actually removed that some time ago. But change got reverted since cloud persistence implementation is returning Internal for transient errors.

Reply:
Oh right, that's what happened. I guess we're working on fixing it eventually?